external-crates/move/crates/move-compiler/src/diagnostics/warning_filters.rs

```diff
@@ -58,8 +61,20 @@ pub type FilterPrefix = Option<Symbol>;
 pub type FilterName = Symbol;
 
 #[derive(PartialEq, Eq, Clone, Debug)]
-pub struct WarningFiltersScope {
-    scopes: Vec<WarningFilters>,
+pub struct WarningFiltersScope(WarningFiltersScope_);
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+enum WarningFiltersScope_ {
+    /// Unsafe and should be used only for internal purposes, such as ide annotations
+    Empty,
+    Static(&'static WarningFilters),
+    Node(Arc<WarningFiltersScopeNode>),
+}
```

Comment:
Switched FiltersScope to a linked list, mostly here to better demonstrate that the initial creation of a `DiagnosticReproter` has to copy/borrow--no clones.